src/renderer/src/routes/modals/ExportFullTransactions/index.tsx

```diff
@@ -0,0 +1,230 @@
+import { useTranslation } from 'react-i18next'
+import { MdDateRange } from 'react-icons/md'
+import { TbChevronRight, TbDeviceFloppy, TbFileExport, TbPackages } from 'react-icons/tb'
+import { ActionStep } from '@renderer/components/ActionStep'
+import { BlockchainIcon } from '@renderer/components/BlockchainIcon'
+import { Button } from '@renderer/components/Button'
+import { DatePicker } from '@renderer/components/DatePicker'
+import { Input } from '@renderer/components/Input'
+import { Separator } from '@renderer/components/Separator'
+import { DateHelper } from '@renderer/helpers/DateHelper'
+import { ToastHelper } from '@renderer/helpers/ToastHelper'
+import { useActions } from '@renderer/hooks/useActions'
+import { useModalNavigate, useModalState } from '@renderer/hooks/useModalRouter'
+import { CenterModalLayout } from '@renderer/layouts/CenterModal'
+import { bsAggregator } from '@renderer/libs/blockchainService'
+import { IAccountState } from '@shared/@types/store'
+import { add, format, sub } from 'date-fns'
+
+type TModalState = {
+  account: IAccountState
+  to: Date
+  from: Date
+}
+
+type TActionData = {
+  account: IAccountState
+  to: Date
+  from: Date
+  selectedFolderPath?: string
+}
+
+export const ExportFullTransactionsModal = () => {
+  const modalState = useModalState<TModalState>()
+  const { modalNavigateWrapper } = useModalNavigate()
+  const { t } = useTranslation('modals', { keyPrefix: 'exportFullTransactions' })
+  const { t: commonT } = useTranslation('common')
+
+  const today = new Date()
+
+  const { actionData, actionState, setData, handleAct } = useActions<TActionData>({
+    account: modalState.account,
+    from: modalState.from ?? sub(today, { weeks: 1 }),
+    to: modalState.to ?? today,
+  })
+
+  const handleSelectAccount = (account: IAccountState) => {
+    setData({ account })
+  }
+
+  const handleBrowse = async () => {
+    const result = await window.api.sendAsync('openDialog', { properties: ['openDirectory', 'createDirectory'] })
+    setData({ selectedFolderPath: result[0] })
+  }
+
+  const handleExport = async () => {
+    try {
+      if (!actionData.selectedFolderPath || !actionData.account.address || !actionData.from || !actionData.to) {
```

Comment:
Create `isDisabled` variable.

Reply:
If I do that, I will need to use non-null assertion operator.